src/frontend/Semantic_check.ml

```diff
@@ -347,6 +348,92 @@ let semantic_check_reduce_sum ~is_cond_dist ~loc id es =
       |> Semantic_error.illtyped_reduce_sum_generic loc id.name
       |> Validate.error
 
+let semantic_check_variadic_ode_tol ~is_cond_dist ~loc id es =
+  let arg_match (x_ad, x_t) y =
```

Comment:
There's some code replication between this function, semantic_check_variadic_ode, and semantic_check_reduce_sum. I think we can at minimum pull out arg_match and args_match

Reply:
Pulled arg match and args_match out of the semantic checks and collapsed the separate _tol and normal check.